### ISSUE TITLE
Tests in test_openmm_equil_rfe_protocols.py fail with pytest=7.4.0

### ISSUE THREAD

**Issue by hannahbaumann** [OpenFreeEnergy/openfe#507]
I created a new conda env from the environment.yml file and got mulitple errors in the test_openmm_equil_rfe_protocols.py, e.g.

file /Users/hannahbaumann/openfe/openfe/tests/protocols/test_openmm_equil_rfe_protocols.py, line 38
  def test_append_topology(benzene_complex_system, toluene_complex_system):
file /Users/hannahbaumann/openfe/openfe/tests/protocols/conftest.py, line 29
  @pytest.fixture
  def benzene_complex_system(benzene_modifications, T4_protein_component):
E       fixture 'benzene_modifications' not found

Downgrading pytest to 7.3.1 seems to fix the issue.

**Comment by IAlibay** [comment]
Yeah I think it was OpenFF TK and then by correspondence ambertools. Should be ok to upgrade now.

**Comment by IAlibay** [comment]
So interestingly py311 is working fine on CI, but I am also about to replicate this 7.4.0 issue on a clean install.

**Comment by dwhswenson** [comment]
1. Is it only with py3.11 and pytest 7.4.0?
2. Is it only for developer installs?
3. What happens if you take the session scoping off of the `benzene_modifications` fixture?

**Comment by IAlibay** [comment]
Only tried on developer installs with 7.4.0 + py3.11, I'll see if I can reproduce later.

My issue was with another fixture (not benzene_modifications, but I didn't note down exactly which one).
